### PR TITLE
Protocol-spec generator should not generate different JSON on each run - Closes #4193

### LINKED ISSUE #4193: Protocol-spec generator should not generate different json on each run
### Description

Generators output different JSON (due to transactions timestamps) on each run and they should always output the same JSON regardless of the time being run

### Acceptance Criteria

- [x] Confirm if transaction timestamps are causing this and fix the date

### PR DESCRIPTION
**Pull request by pablitovicente**
# What was the problem?

A timestamp was not being set so generated transaction would always have a different signature and id.

### How did I solve it?

By fixing a configurable timestamp for all transactions.

### How to manually test it?

Run the generators several times and compare the outputs

### Review checklist

- [x] The PR resolves #4193
- [x] All new code is covered with unit tests
- [x] Relevant integration / functional tests are added
- [x] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [x] Documentation has been added/updated

### CODE REVIEW

File: protocol-specs/generators/base_generator.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ const runGenerator = (generatorName, suiteCreators) => {
 
 		const fullPath = `${dirPath}/${suiteResult.handler}.json`;
 
-		fs.writeFileSync(fullPath, JSON.stringify(suiteResult, null, 2));
+		fs.writeFileSync(fullPath, `${JSON.stringify(suiteResult, null, 2)}\u000A`);
```
Comment (shuse2) [inline protocol-specs/generators/base_generator.js]: why do we need to do this way? maybe comment would be nice

File: protocol-specs/generators/block_processing_second_signature/index.js [hunk 2]
```diff
@@ -292,12 +303,16 @@ const generateTestCasesInvalidBlockSecondSignatureTxSecondTime = () => {
 
 	const secondSignature =
 		'erupt sponsor rude supreme vacant delay salute allow laundry swamp curve brain';
+	const secondPassphraseObject = registerSecondPassphrase({
```
Comment (shuse2) [inline protocol-specs/generators/block_processing_second_signature/index.js]: do we need to use this function at all?

File: protocol-specs/generators/block_processing_second_signature/index.js [hunk 3]
```diff
@@ -200,22 +207,26 @@ const generateTestCasesValidBlockSecondSignatureTx = () => {
...
 const generateTestCasesinvalidBlockWithSecondSignatureAndFundsTxSuite = () => {
 	const amount = '5500000000';
-	const transferTx = new TransferTransaction(
-		transfer({
-			amount,
-			passphrase: accounts.genesis.passphrase,
-			recipientId: accounts.existingDelegate.address,
-		}),
-	);
+	const transferObject = transfer({
```
Comment (shuse2) [inline protocol-specs/generators/block_processing_second_signature/index.js]: same as below. Do we need to mix this function and the instance?
It was meant to be used one way or another

Comment (pablitovicente) [reply protocol-specs/generators/block_processing_second_signature/index.js]: It was faster to implement this way with the least amount of changes I will take the extra time for doing it from JSON/Object literal to instance.

File: protocol-specs/generators/base_generator.js (hunk 1 above)
Comment (pablitovicente) [reply protocol-specs/generators/base_generator.js]: Prettier was removing last line and linter was complaining about \n will look for workarounds or add comment.

Comment (pablitovicente) [reply protocol-specs/generators/base_generator.js]: Fixed guess was my IDE being funky

File: protocol-specs/generators/block_processing_second_signature/index.js (hunk 3 above)
Comment (pablitovicente) [reply protocol-specs/generators/block_processing_second_signature/index.js]: Fixed.

File: protocol-specs/generators/block_processing_second_signature/index.js (hunk 2 above)
Comment (pablitovicente) [reply protocol-specs/generators/block_processing_second_signature/index.js]: Removed and replaced by object and `getKeys`

**Pull request merged**